core/mapper/exprmapper/function/array/length/length.go

```diff
@@ -0,0 +1,33 @@
+package length
+
+import (
+	"github.com/TIBCOSoftware/flogo-lib/core/data"
+	"github.com/TIBCOSoftware/flogo-lib/core/mapper/exprmapper/expression/function"
+	"github.com/TIBCOSoftware/flogo-lib/logger"
+)
+
+var log = logger.GetLogger("length-function")
+
+type Length struct {
+}
+
+func init() {
+	function.Registry(&Length{})
+}
+
+func (s *Length) GetName() string {
+	return "length"
+}
+
+func (s *Length) GetCategory() string {
+	return "array"
+}
+
+func (s *Length) Eval(arr interface{}) int {
+	log.Debugf("Return the length of array \"%v\"", arr)
+	myArr, err := data.CoerceToArray(arr)
+	if err != nil {
```

Comment:
we should return the error out.  define function return with (int, error).  otherwise, it might confuse that return length 0 while error occurred.